### ISSUE TITLE
Add XML as a Data File Type

### ISSUE THREAD

**Issue by bzerangue** [gohugoio/hugo#4470]
It would be nice to provide XML as an available data type to load. 

Currently, Hugo accepts, TOML, YAML, JSON, and CSV. Is there any reason why XML is NOT amongst the available formats?

It would be great to add XML as an available data type for the `/data` folder.

**Comment by rdwatters** [comment]
@bzerangue This would be fantastic. Not sure if something like https://github.com/beevik/etree would help said efforts...

**Comment by fbaube** [comment]
+1 on this. Type "XML" could then act as a flag to invoke DOCTYPE parsing and optional validation.

**Comment by kaushalmodi** [comment]
Having a `getXML` similar to `getJSON` would be useful in creating "planets" from feeds. Ref: https://discourse.gohugo.io/t/anyone-here-interested-in-a-hugo-planet/15092.

**Comment by stephlocke** [comment]
This'll be great for surfacing RSS from other sources like Medium

**Comment by raybellis** [comment]
+1 from me, too - I'd like to be able to include references from the RFC index, but it's only published in XML format.

**Comment by fbaube** [comment]
I'm working on a Go library to parse mixed content. If anyone can propose an API (i.e. interfaces) that would be useful for Hugo, I could give it a whack.

**Comment by stephlocke** [comment]
On the RSS specific side, I currently leverage a site that parses RSS to JSON for consumption

```
{{ $rssJ := getJSON "https://api.rss2json.com/v1/api.json?rss_url=https%3A%2F%2Fmedium.com%2Ffeed%2Fnightingale-hq" }}


{{ range $rssJ.items }}
    {{ $post := . }}
    <div class="col-12 col-md-6 col-lg-4 mb-2 mr-2 pb-8 blog" style="background-image: url({{ $post.thumbnail }});">
    <a href="{{ $post.link }}" target="_blank" alt="{{ $post.title }}">
    <h2>{{ $post.title }}</h2>
    </a>
    <h3>{{ $post.author }}</h3>
    </div>
{{ end }}
```

**Comment by felicianotech** [comment]
If this request is too complex as is, what about simplifying it to a `getRSS` function? I know for me the need for `getXML` is simply so that I can have Hugo read in RSS feeds.

**Comment by lhenze** [comment]
Agreed, this would be fantastic.  From the library tech perspective --  we are experimenting with using Hugo to generate Special Collections Finding Aids from Encoded Archival Description (EAD) -- an XML format widely used by libraries.

**Comment by vanbroup** [comment]
#9044 is another attempt to get XML data support

While this implementation provides less flexibility than my previous implementation using `xmlquery` (#9031), this one doesn't come with its own API and is much easier to use (equal to `getJSON`):

Sample usage:
```
{{ with getXML "https://www.w3schools.com/xml/note.xml" }}
{{ .note.body }}
{{ end }}
```

A common usage:
```
{{ with getXML "https://example.com/feed.rss" }}
{{ range .rss.channel.item }}
    <strong>{{ .title | plainify | htmlUnescape }}</strong><br />
    <p>{{ .description | plainify | htmlUnescape }}</p>
    {{ $link := .link | plainify | htmlUnescape }}
    <a href="{{ $link }}">{{ $link }}</a><br />
    <hr>
{{ end }}
{{ end }}
```

**Comment by swamidass** [comment]
It would also be good for reading RSS feeds.
